src/bin/rav1e.rs

```diff
@@ -157,7 +159,21 @@ fn process_frame<T: Pixel, D: Decoder>(
         {
           write_y4m_frame(y4m_enc_uw, rec, y4m_details);
         }
-        return Ok(Some(pkt.into()));
+        let metrics_config = source.input.get_video_details();
+        let shared_frame = Arc::new(&pkt);
+        let metrics_input_frame: &Frame<T> =
+          shared_frame.deref().source.as_ref().unwrap();
+        let metrics_output_frame: &Frame<T> =
+          shared_frame.deref().rec.as_ref().unwrap();
+        let metrics_cli = parse_cli().unwrap().metrics_enabled;
+        encode_metrics = calculate_frame_metrics(
+          metrics_input_frame,
+          metrics_output_frame,
+          metrics_config.bit_depth,
+          metrics_config.chroma_sampling,
+          metrics_cli,
+        );
+        return caclulate_frame_summary(pkt, encode_metrics);
```

Comment:
Still has a typo. The whole code block above can be moved in `calculate_frame_summary`.